src/EditorFeatures/Core/InlineRename/UI/Adornment/RenameFlyout.xaml.cs

```diff
@@ -131,30 +132,47 @@ private void TextView_LayoutChanged(object sender, TextViewLayoutChangedEventArg
 
     private void PositionAdornment()
     {
-        var span = _viewModel.InitialTrackingSpan.GetSpan(_textView.TextSnapshot);
-        var line = _textView.GetTextViewLineContainingBufferPosition(span.Start);
-        var charBounds = line.GetCharacterBounds(span.Start);
+        try
+        {
+            // Get the current text snapshot and validate the tracking span is still valid.
+            // In scenarios with multiple document groups, the text buffer can be replaced
+            // and the tracking span becomes invalid, causing exceptions when trying to map
+            // the span to the new snapshot.
+            var currentSnapshot = _textView.TextSnapshot;
+            var span = _viewModel.InitialTrackingSpan.GetSpan(currentSnapshot);
+
+            var line = _textView.GetTextViewLineContainingBufferPosition(span.Start);
+            var charBounds = line.GetCharacterBounds(span.Start);
 
-        var height = DesiredSize.Height;
-        var width = DesiredSize.Width;
+            var height = DesiredSize.Height;
+            var width = DesiredSize.Width;
 
-        var desiredTop = charBounds.TextBottom + 5;
-        var desiredLeft = charBounds.Left;
+            var desiredTop = charBounds.TextBottom + 5;
+            var desiredLeft = charBounds.Left;
 
-        var top = (desiredTop + height) > _textView.ViewportBottom
-            ? _textView.ViewportBottom - height
-            : desiredTop;
+            var top = (desiredTop + height) > _textView.ViewportBottom
+                ? _textView.ViewportBottom - height
+                : desiredTop;
 
-        var left = (desiredLeft + width) > _textView.ViewportRight
-            ? _textView.ViewportRight - width
-            : desiredLeft;
+            var left = (desiredLeft + width) > _textView.ViewportRight
+                ? _textView.ViewportRight - width
+                : desiredLeft;
 
-        MaxWidth = _textView.ViewportRight;
-        MinWidth = Math.Min(DefaultMinWidth, _textView.ViewportWidth);
+            MaxWidth = _textView.ViewportRight;
+            MinWidth = Math.Min(DefaultMinWidth, _textView.ViewportWidth);
 
-        // Top can be negative if the viewport is scrolled up, but not left
-        Canvas.SetTop(this, top);
-        Canvas.SetLeft(this, Math.Max(0, left));
+            // Top can be negative if the viewport is scrolled up, but not left
+            Canvas.SetTop(this, top);
+            Canvas.SetLeft(this, Math.Max(0, left));
+        }
+        catch (Exception ex)
+        {
+            // The tracking span may no longer be valid for the current text snapshot.
+            // This can happen when working with multiple document groups where the
+            // text buffer has been replaced. Dismiss the rename flyout gracefully.
+            FatalError.ReportAndCatch(ex);
```

Comment:
This is expected to go in an exception filter so it can report before the stack has unwound.